Mol_Analysis.py

```diff
@@ -266,31 +267,53 @@ def mkndx(self, commands, ffile=None, **kwargs):
             print("Making ndx file")
             g_mkndx.run(**kwargs)
 
-    def mindist(self, **kwargs):
-        """Runs gromacs mindist on the new, centered traj."""
-        outfile = self.odir + "min_pbc_dist.xvg"
-        self._addfilename("mindist", outfile)
-        g_mindist = gromacs.tools.G_mindist(
-                f=self.outf,
-                s=self.tpr,
-                pi=True,
-                od=outfile,
-                input="Protein",
-                stdout=False,
-                stderr=False
-                )
 
-        if not self.dryrun:
-            print("Running mindist")
-            excode, out, err = g_mindist.run(**kwargs)
-        if not self.dryrun:
-            for line in out.split("\n"):
-                sline = line.split()
-                if len(sline) > 2:
-                    if ["The", "shortest", "periodic"] == sline[0:3]:
-                        self.minpdist = float(sline[5])
-                        self.mintime = float(sline[9])
-                        break
+    def mindist(self, num_cores=10, **kwargs):
+        """Merges output from mindist_slice, keeping header only once and removing duplicate rows."""
+        
+        # Determine chunk size based on total simulation length and split
+        if not hasattr(self, 'simlen') or not hasattr(self, 'trjdt'):
+            self.getinfo(self.outf)  # Populate simlen and trjdt attributes
+        
+        chunk_size = self.simlen // num_cores
+        splits = [(i, min(i + chunk_size, self.simlen)) for i in range(0, self.simlen, chunk_size)]
+        print(f"Total time slices for mindist: {len(splits)}")
+
+        # Run mindist_slice in parallel to generate individual .xvg files
+        pool = mp.Pool(processes=num_cores)
+        args = [(begin, end, self.outf, self.tpr, self.odir, self.dryrun) for begin, end in splits]
+        xvg_files = [f for f in pool.map(mindist_slice, args) if f is not None]
```

Comment:
this sound a case where `pool.starmap()` would come handy - so that your function can accept the actual arguments it needs and so it's easier to test/run and develop with